auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java

```diff
@@ -77,7 +83,46 @@ protected void onCreate(Bundle savedInstanceState) {
         mSaveSmartLock = mActivityHelper.getSaveSmartLockInstance();
         findViewById(R.id.email_provider).setOnClickListener(this);
 
-        populateIdpList(mActivityHelper.getFlowParams().providerInfo);
+        Map<String, IdpConfig> providerIdToConfig = new HashMap<>();
+        List<IdpConfig> allConfiguredProviders = mActivityHelper.getFlowParams().providerInfo;
+        for (IdpConfig providerConfig : allConfiguredProviders) {
+            providerIdToConfig.put(providerConfig.getProviderId(), providerConfig);
+        }
+        List<IdpConfig> visibleProviders = new ArrayList<>();
+        if (mActivityHelper.getFlowParams().isReauth) {
+            // display the reauthReason if available
+            String reauthReason = mActivityHelper.getFlowParams().reauthReason;
+            if (!TextUtils.isEmpty(reauthReason)) {
+                TextView reauthReasonTextView = (TextView) findViewById(R.id.reauth_reason);
+                reauthReasonTextView.setText(reauthReason);
+                reauthReasonTextView.setVisibility(View.VISIBLE);
+            }
+
+            // For reauth flow we only want to show the IDPs which the user has associated with
+            // their account.
+            List<String> providerIds = mActivityHelper.getCurrentUser().getProviders();
+            if (providerIds.size() == 0) {
+                // zero providers indicates that it is an email account
```

Comment:
Doesn't the providers array contain just `"password"` in this case or is that only when you call `fetchProvidersForEmail()`?